src/common/Pagination/index.js

```diff
@@ -1,70 +1,157 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { selectMovies } from '../../features/movies/moviesSlice';
+import { selectPopularPeople } from '../../features/people/popularPeopleSlice';
 import Button from "./Button";
 import Icon from "./Icon";
 import PageInfo from './PageInfo';
 import { Footer } from './styled';
 
 const Pagination = ({ type }) => {
+
+  const movies = useSelector(selectMovies);
+  const people = useSelector(selectPopularPeople);
+  const params = useParams();
+
+  const [page, setPage] = useState(+params.page);
+  const [backButtonState, setBackButtonState] = useState(false);
+  const [forwardButtonState, setForwardButtonState] = useState(false);
+
+  const backButtonStateHandler = () => {
+    if (type === "movies") {
+      if (page === 1) {
+        setBackButtonState(true);
+      } else
+        setBackButtonState(false);
+    };
+    if (type === "people") {
+      if (page === 1) {
+        setBackButtonState(true);
+      } else
+        setBackButtonState(false);
+    };
+  };
+
+  const forwardButtonStateHandler = () => {
+    if (type === "movies") {
+      if (page === movies.total_pages) {
+        setForwardButtonState(true);
+      } else
+        setForwardButtonState(false);
+    };
+    if (type === "people") {
+      if (page === people.total_pages) {
+        setForwardButtonState(true);
+      } else
+        setForwardButtonState(false);
+    };
+  };
+
+  useEffect(() => {
+    backButtonStateHandler();
+    forwardButtonStateHandler();
+    // eslint-disable-next-line
```

Comment:
co to za komentarz?

Reply:
a żeby linter nie wywalał błędu w konsoli i ja mam wtyczkę, że pokazuje mi błędy też od razu w edytorze i mi to przeszkadzało bo useEffect dopomina się dodania tych funkcji do tablic zależności a zrobienie tego to powoduje kolejne błędy.
zapomniałem usunąć